helix-vcs/src/git.rs

```diff
@@ -61,6 +74,28 @@ impl Git {
 
         Ok(res)
     }
+
+    /// Emulates the result of running `git status` from the command line.
+    fn status(repo: &Repository) -> Result<Box<dyn Iterator<Item = Result<FileChange>>>> {
+        let work_dir = repo
+            .work_dir()
+            .ok_or_else(|| anyhow::anyhow!("working tree not found"))?
+            .to_path_buf();
+
+        // Here we discard the `status.showUntrackedFiles` config, as it makes little sense in our
+        // case to not list new (untracked) files. We could have respected this config if the
+        // default value weren't `Collapsed` though, as this default value would render the feature
+        // unusable to many.
+        let status_platform = repo
+            .status(gix::progress::Discard)?
+            .untracked_files(UntrackedFiles::Files)
+            .index_worktree_rewrites(Some(Default::default()));
```

Comment:
It's probably better to bring in the `Rewrite` structure in order to make clear what's going on. Even I don't know actually 😅 - my guess is that it's on with identity based rename tracking.

Reply:
Now explicitly constructing the `Rewrites` struct.